app/assets/javascripts/discourse/app/initializers/codeblock-buttons.js

```diff
@@ -0,0 +1,44 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+import CodeblockButtons from "discourse/lib/codeblock-buttons";
+
+let _codeblockButtons = [];
+
+export default {
+  name: "codeblock-buttons",
+
+  initialize(container) {
+    const siteSettings = container.lookup("site-settings:main");
+
+    withPluginApi("0.8.7", (api) => {
```

Comment:
Why 0.8.7?

Reply:
Also interesting, we're using the pluginApi even though we're in core. Is that because it's just convenient, rather than having to insert something into the post itself?

Reply:
It's just the original code, will update to use the latest plugin API. As for this:

> Also interesting, we're using the pluginApi even though we're in core. Is that because it's just convenient, rather than having to insert something into the post itself?

Yes I believe it's because it's convinient and we don't have to add something to the post itself. We do this a bunch of times in other places in core too.